bkpr-listaccountevents filter by id

michael1011 opened the pull request:
Closes #7518

It's much nicer to ask the database to filter than getting *everything* and filtering the output.

```
time ./cli/lightning-cli --regtest -k bkpr-listaccountevents | grep x 
         "payment_id": "x",

real	0m6,908s
user	0m1,745s
sys	0m0,534s
```

```
time ./cli/lightning-cli --regtest -k bkpr-listaccountevents payment_id=x 
{
   "events": [
      {
         "payment_id": "x",
         <omitted data>
      }
   ]
}

real	0m0,074s
user	0m0,002s
sys	0m0,000s
```

niftynei commented on plugins/bkpr/bookkeeper.c:
```diff
@@ -386,6 +395,16 @@ static struct command_result *json_list_account_events(struct command *cmd,
+	} else if (payment_id != NULL) {
+		channel_events = get_channel_events_by_id(cmd, db, payment_id);
+
+		// Transaction ids are stored as big-endian in the database
```
nit: comments in cln are always done as /* block */ style

niftynei commented on plugins/bkpr/bookkeeper.c:
```diff
@@ -386,6 +395,16 @@ static struct command_result *json_list_account_events(struct command *cmd,
+		// Transaction ids are stored as big-endian in the database
+		reverse_bytes(payment_id->u.u8, sizeof(payment_id->u.u8));
+		tx_id = tal(cmd, struct bitcoin_txid);
+		tx_id->shad.sha = *payment_id;
```
nit: put the payment_id data into the tx_id, and then reverse the bytes. this leaves the `payment_id` data untouched/true to self.

niftynei reviewed (commented):
ACK bb328772d79783ebf87a2f50f5b079866cb473ec

two tiny nits that would be nice to fixup but largely optional imo. i think the CI error is a global/master one?

michael1011 commented:
Addressed the comments.

> i think the CI error is a global/master one?

It seems to me like CI is little flaky. Let's see if the new commit passes that step
